components/profile_popover/profile_popover.tsx

```diff
@@ -16,263 +13,270 @@ import {browserHistory} from 'utils/browser_history';
 import Pluggable from 'plugins/pluggable';
 import AddUserToChannelModal from 'components/add_user_to_channel_modal';
 import LocalizedIcon from 'components/localized_icon';
 import ToggleModalButtonRedux from 'components/toggle_modal_button_redux';
 import Avatar from 'components/widgets/users/avatar';
 import Popover from 'components/widgets/popover';
-import SharedUserIndicator from 'components/shared_user_indicator.tsx';
+import SharedUserIndicator from 'components/shared_user_indicator';
 import CustomStatusEmoji from 'components/custom_status/custom_status_emoji';
 import CustomStatusModal from 'components/custom_status/custom_status_modal';
 import CustomStatusText from 'components/custom_status/custom_status_text';
+import {UserCustomStatus, UserProfile, UserTimezone} from 'mattermost-redux/types/users';
+import {Dictionary} from 'mattermost-redux/types/utilities';
 
 import './profile_popover.scss';
 
+interface ProfilePopoverProps extends Omit<React.ComponentProps<typeof Popover>, 'id'>{
+
+    /**
+     * Source URL from the image to display in the popover
+     */
+    src: string;
+
+    /**
+     * Source URL from the image that should override default image
+     */
+    overwriteIcon?: React.ReactNode;
+
+    /**
+     * User the popover is being opened for
+     */
+    user?: Partial<UserProfile>;
+    userId: string;
+    channelId?: string;
+
+    /**
+     * Status for the user, either 'offline', 'away', 'dnd' or 'online'
+     */
+    status?: string;
+    hideStatus?: boolean;
+
+    /**
+     * Function to call to hide the popover
+     */
+    hide?: () => void;
+
+    /**
+     * Set to true if the popover was opened from the right-hand
+     * sidebar (comment thread, search results, etc.)
+     */
+    isRHS?: boolean;
+    isBusy?: boolean;
+
+    /**
+     * Returns state of modals in redux for determing which need to be closed
+     */
+    modals?: {
+        [modalId: string]: {
+            open: boolean;
+            dialogProps: Dictionary<any>;
+            dialogType: React.Component;
+        };
+    };
+    currentTeamId: string;
+
+    /**
+     * @internal
+     */
+    currentUserId: string;
+    customStatus?: UserCustomStatus | null;
+    isCustomStatusEnabled: boolean;
+
+    /**
+     * @internal
+     */
+    hasMention?: boolean;
+
+    /**
+     * @internal
+     */
+    isInCurrentTeam: boolean;
+
+    /**
+     * @internal
+     */
+    teamUrl: string;
+
+    /**
+     * @internal
+     */
+    isTeamAdmin: boolean;
+
+    /**
+     * @internal
+     */
+    isChannelAdmin: boolean;
+
+    /**
+     * @internal
+     */
+    canManageAnyChannelMembersInCurrentTeam: boolean;
+
+    /**
+     * The overwritten username that should be shown at the top of the popover
+     */
+    overwriteName?: React.ReactNode;
+
+    /**
+     * @internal
+     */
+    enableTimezone: boolean;
+    actions: {
+        openModal: (modalData: {modalId: string; dialogType: any; dialogProps?: any}) => Promise<{
```

Comment:
`dialogType` here should be `React.ComponentType`

Reply:
@devinbinnie Converting to React.ComponentType throwing error of the wrong type. I can see everywhere we have used as any only. Could please share any reference to check

Reply:
Can you try `React.Component`?

Reply:
Yeah I did

Reply:
Have checked to assign Type as React.Component or React.ComponentType throw type error as we are assign object of different type inside class

Reply:
I wouldn't be surprised if there's some weird typing issues going on here, but it looks like we use `React.Component` in some places
https://github.com/mattermost/mattermost-webapp/blob/master/components/sidebar/sidebar_category/sidebar_category_menu/index.ts#L28 
and any in others
https://github.com/mattermost/mattermost-webapp/blob/master/components/sidebar/sidebar_category/sidebar_category_menu/sidebar_category_menu.tsx#L26

Notably though, it seems like we use `React.Component` only in index files currently and `any` in the components themselves, so we might be hitting some weirdness with our redux typing.

Reply:
<img width="824" alt="Screenshot 2021-06-07 at 9 33 57 PM" src="https://user-images.githubusercontent.com/16203333/121052594-26a2d100-c7d8-11eb-9afb-af250877da73.png">

Reply:
@hmhealey @devinbinnie This is the type getting inferred by typescript of the UsersettingModal

Reply:
Given the above comments, we can just keep this as `any` for now, though it would be worth looking into why this doesn't work.

Reply:
Okay

Reply:
I will check if I can find some good reason